build/int.cloudbuild.yaml

```
@@ -14,110 +14,148 @@
 
 timeout: 12600s
 steps:
+
 - id: prepare
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
   args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && prepare_environment']
   env:
   - 'TF_VAR_folder_id=$_FOLDER_ID'
   - 'TF_VAR_billing_account=$_BILLING_ACCOUNT'
   - 'TF_VAR_group_email=test-gcp-org-admins@test.infra.cft.tips'
-- id: prepare-network-hub-and-spoke-config
+
+- id: sleep
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh &&  ./test/network_test_prepare.sh -p']
-- id: create
+  args: ['/bin/bash', '-c', 'sleep 240']
+
+- id: create-bootstrap
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do create']
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && RUN_STAGE=init go test -v ./... -p 1 -timeout 0 -run ^TestBootstrap$']
```

Was there any issue using the CFT cli in cloudbuild?
```suggestion
  args: ['/bin/bash', '-c', 'cft test run TestBootstrap --stage init --verbose']
```

---

I got this error 

```bash
Step #2 - "create-bootstrap": t=2022-04-15T03:59:37+0000 lvl=info msg="using test-dir: ."
Step #2 - "create-bootstrap": t=2022-04-15T03:59:37+0000 lvl=warn msg="Skipping discovered test. discover_test.go not found."
Step #2 - "create-bootstrap": Error: unable to find TestBootstrap- one of ["all"] expected
Step #2 - "create-bootstrap": Usage:
Step #2 - "create-bootstrap":   cft test run [flags]
Step #2 - "create-bootstrap":   
Step #2 - "create-bootstrap": 
Step #2 - "create-bootstrap": Flags:
Step #2 - "create-bootstrap":   -h, --help           help for run
Step #2 - "create-bootstrap":       --stage string   Test stage to execute (default is running all stages in order - init, apply, verify, teardown)
Step #2 - "create-bootstrap": 
Step #2 - "create-bootstrap": Global Flags:
Step #2 - "create-bootstrap":       --test-dir string   Path to directory containing integration tests (default is computed by scanning current working directory)
Step #2 - "create-bootstrap":       --verbose           Log output to stdout
Step #2 - "create-bootstrap": 
```

when using

```yaml
- id: create-bootstrap
  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && cd test/integration && cft test run TestBootstrap --stage init --verbose']
```

running `cft test run TestBootstrap --stage init --verbose` from the inside the image with `make docker_run` worked

I will test with your suggestion

---

testing with

```
- id: create-bootstrap
  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
  args: ['/bin/bash', '-c', 'cft test run TestBootstrap --stage init --verbose']
```

gives the same error

```
Already have image (with digest): gcr.io/cloud-foundation-cicd/cft/developer-tools:1.4
t=2022-04-23T04:00:15+0000 lvl=info msg="using test-dir: ."
Error: unable to find TestBootstrap- one of ["all"] expected
t=2022-04-23T04:00:15+0000 lvl=warn msg="Skipping discovered test. discover_test.go not found."
Usage:
  cft test run [flags]
  

Flags:
  -h, --help           help for run
      --stage string   Test stage to execute (default is running all stages in order - init, apply, verify, teardown)

Global Flags:
      --test-dir string   Path to directory containing integration tests (default is computed by scanning current working directory)
      --verbose           Log output to stdout

```

---

It works if we set the `test-dir` param

```yaml
- id: create-bootstrap
  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
  args: ['/bin/bash', '-c', 'cft test run TestBootstrap --stage init --verbose --test-dir /workspace/test/integration']

```

```shell
t=2022-04-25T12:33:43+0000 lvl=info msg="using test-dir: /workspace/test/integration"
t=2022-04-25T12:33:43+0000 lvl=warn msg="Skipping discovered test. discover_test.go not found."
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/app-infra: unable to find config in /workspace/test/fixtures/app-infra nor /workspace/examples/app-infra"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/bootstrap: unable to find config in /workspace/test/fixtures/bootstrap nor /workspace/examples/bootstrap"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/envs: unable to find config in /workspace/test/fixtures/envs nor /workspace/examples/envs"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/networks: unable to find config in /workspace/test/fixtures/networks nor /workspace/examples/networks"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/org: unable to find config in /workspace/test/fixtures/org nor /workspace/examples/org"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/projects: unable to find config in /workspace/test/fixtures/projects nor /workspace/examples/projects"
t=2022-04-25T12:33:43+0000 lvl=warn msg="unable to discover configs for /workspace/test/integration/shared: unable to find config in /workspace/test/fixtures/shared nor /workspace/examples/shared"
t=2022-04-25T12:33:43+0000 lvl=dbug msg="running /usr/local/bin/go with args [go test ./bootstrap -run TestBootstrap -p 1 -count 1 -timeout 0 -v] in /workspace/test/integration"
go: downloading github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test v0.0.0-20211001192917-5e783cf7c716
```

---

@daniel-cit yeah since we dont discover_test.go its unable to auto-discover it. We can add the --test-dir flag and lets also open an issue to discover it from an env var like BPT_TEST_DIR.

---

issue opend
https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/issues/1126